src/components/ec-timer/ec-timer.vue

```diff
@@ -46,96 +46,87 @@
 </template>
 
-<script>
+<script setup>
+import {
+  computed, onBeforeUnmount, ref, watchEffect,
+} from 'vue';
+
 import Countdown from '../../utils/countdown';
 
-export default {
-  name: 'EcTimer',
-  props: {
-    /**
-     * Seconds to finish the countdown
-    */
-    seconds: {
-      type: Number,
-      required: true,
-      validator(value) {
-        return Number.isInteger(value) && value > 0;
-      },
-    },
-    /**
-     * Indicates if the countdown is running
-    */
-    isRunning: {
-      type: Boolean,
-      required: true,
+const props = defineProps({
+  /**
+   * Seconds to finish the countdown
+   */
+  seconds: {
+    type: Number,
+    required: true,
+    validator(value) {
+      return Number.isInteger(value) && value > 0;
     },
   },
-  emits: ['time-expired'],
-  data() {
-    return {
-      radius: 24,
-      strokeWidth: 4,
-      secondsLeft: this.seconds,
-      startTime: null,
-      currentTime: null,
-      countdown: null,
-    };
+  /**
+   * Indicates if the countdown is running
+   */
+  isRunning: {
+    type: Boolean,
+    required: true,
   },
-  computed: {
-    diameter() {
-      return (this.radius * 2) + this.strokeWidth;
-    },
-    viewbox() {
-      return `0 0 ${this.diameter} ${this.diameter}`;
-    },
-    circumference() {
-      return 2 * Math.PI * this.radius;
-    },
-    steps() {
-      return this.circumference / this.seconds;
-    },
-    offset() {
-      return this.circumference + this.steps * this.secondsLeft;
-    },
-  },
-  watch: {
-    isRunning: {
-      immediate: true,
-      handler(value) {
-        if (value) {
-          this.startCountdown();
-        } else {
-          this.stopCountdown();
-          this.secondsLeft = this.seconds;
-        }
-      },
-    },
-  },
-  beforeUnmount() {
-    this.stopCountdown();
-  },
-  methods: {
-    startCountdown() {
-      this.countdown = new Countdown();
-      this.countdown.start(this.seconds);
-      this.countdown.on('time-updated', (secondsLeft) => {
-        this.secondsLeft = secondsLeft;
-      });
-      this.countdown.on('time-expired', () => {
-        /**
-         * Emitted after the countdown is finish
-         * @event time-expired
-         * @type {void}
-        */
-        this.$emit('time-expired');
-      });
-    },
-    stopCountdown() {
-      if (this.countdown) {
-        this.countdown.stop();
-        this.countdown = null;
-      }
-    },
+});
+const emit = defineEmits(['time-expired']);
+
+const diameter = computed(() => radius.value * 2 + strokeWidth.value);
+const viewbox = computed(() => `0 0 ${diameter.value} ${diameter.value}`);
+const circumference = computed(() => 2 * Math.PI * radius.value);
+const steps = computed(() => circumference.value / props.seconds);
+const offset = computed(() => circumference.value + steps.value * secondsLeft.value);
+
+const radius = ref(24);
+const strokeWidth = ref(4);
+const secondsLeft = ref(props.seconds);
+const countdown = ref(null);
```

Comment:
Do we need the countdown to be reactive? is it used anywhere in the template or computed props? We can save some work for Vue's reactivity and just do this without `ref`. We don't need to then `countdown.value` anywhere but just `countdown`.

Reply:
Agreed, it can just be a basic const. As its not in the template nor a computed function.

Reply:
changed it to be a let